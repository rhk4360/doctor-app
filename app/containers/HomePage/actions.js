/*
 * Home Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 */

import {
  CHANGE_USERNAME,
  CHANGE_PASSWORD,
  SIGN_IN,
  SIGN_IN_SUCCESS,
  SIGN_IN_ERROR,
} from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {name} name The new text of the input field
 *
 * @return {object}    An action object with a type of CHANGE_USERNAME
 */
export function changeUsername(name) {
  return {
    type: CHANGE_USERNAME,
    name,
  };
}

/**
 * Changes the input field of the form
 *
 * @param  {password} password The new text of the input field
 *
 * @return {object}    An action object with a type of CHANGE_PASSWORD
 */
export function changePassword(password) {
  return {
    type: CHANGE_PASSWORD,
    password,
  };
}

/**
 * Attempts to sign the user in
 *
 * @return {object}    An action object with a type of SIGN_IN
 */
export function signIn() {
  return {
    type: SIGN_IN,
  };
}

/**
 * Successful sign in
 *
 * @param  {user} user that signed in
 *
 * @return {object}    An action object with a type of SIGN_IN_SUCCESS
 */
export function signInSuccess(user) {
  return {
    type: SIGN_IN_SUCCESS,
    user,
  };
}

/**
 * Error during sign in
 *
 * @return {object}    An action object with a type of SIGN_IN_ERROR
 */
export function signInError() {
  return {
    type: SIGN_IN_ERROR,
  };
}
