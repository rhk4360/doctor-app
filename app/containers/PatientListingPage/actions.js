
import {
  GET_PATIENT_LIST,
  GET_PATIENTS_SUCCESS,
  GET_PATIENTS_ERROR,
} from './constants';

/**
 * Gets a list of patients 
 *
 *
 * @return {object}    An action object with a type of FETCH_PATIENT_LIST
 */
export function getPatientsList() {
  console.log('retrievePatients');
  return {
    type: GET_PATIENT_LIST,
  };
};

/**
 * Successfully retrieved a list of patients 
 *
 * @param {array} - array of patients
 *
 * @return {object}    An action object with a type of GET_PATIENTS_SUCCESS
 */
export function getPatientsSuccess(list) {
  return {
    type: GET_PATIENTS_SUCCESS,
    list: list,
  };
};

/**
 * An error occurred while trying to fetch patients 
 *
 * @param {object} - error that occurred
 *
 * @return {object}    An action object with a type of GET_PATIENTS_ERROR
 */
export function getPatientsError(error) {
  return {
    type: GET_PATIENTS_ERROR,
    error: error,
  };
};

