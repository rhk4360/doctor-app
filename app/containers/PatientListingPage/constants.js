export const GET_PATIENT_LIST = 'boilerplate/PatientListing/GET_PATIENT_LIST';
export const GET_PATIENTS_SUCCESS = 'boilerplate/PatientListing/GET_PATIENTS_SUCCESS';
export const GET_PATIENTS_ERROR = 'boilerplate/PatientListing/GET_PATIENTS_ERROR';
