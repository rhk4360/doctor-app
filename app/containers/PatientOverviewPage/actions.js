import {
  GET_PATIENT_OVERVIEW,
  GET_PATIENT_OVERVIEW_SUCCESS,
  GET_PATIENT_OVERVIEW_ERROR,
  CANCEL_APPOINTMENT,
  UPDATE_APPOINTMENT_SUCCESS,
  UPDATE_APPOINTMENT_ERROR,
  DECLINE_APPOINTMENT,
  DECLINE_APPOINTMENT_SUCCESS,
  DECLINE_APPOINTMENT_ERROR,
} from './constants';

/**
 * Gets a patient's details based on id
 *
 * @param id - patient's id in the system
 *
 * @return {object}    An action object with a type of GET_PATIENT_OVERVIEW
 */
export function getPatientOverview(id) {
  return {
    type: GET_PATIENT_OVERVIEW,
    id,
  };
};

/**
 * Successfully retrieved a patient overview 
 *
 * @param {object} - patient details
 *
 * @return {object}    An action object with a type of GET_PATIENT_OVERVIEW_SUCCESS
 */
export function getPatientOverviewSuccess(overview) {
  return {
    type: GET_PATIENT_OVERVIEW_SUCCESS,
    overview,
  };
};

/**
 * An error occurred while trying get patient details
 *
 * @param {object} - error that occurred
 *
 * @return {object}    An action object with a type of GET_PATIENT_OVERVIEW_ERROR
 */
export function getPatientOverviewError(error) {
  return {
    type: GET_PATIENT_OVERVIEW_ERROR,
    error,
  };
};

/**
 * Cancel an appointment
 *
 * @param appointmentToUpdate - appointment to update
 *
 * @return {object}    An action object with a type of CANCEL_APPOINTMENT
 */
export function cancelAppointment(appointmentToUpdate) {
  return {
    type: CANCEL_APPOINTMENT,
    appointmentToUpdate,
  }
};

/**
 * Canceled an appointment successfully
 *
 * @param updatedAppointment - newly updated appointment
 *
 * @return {object}    An action object with a type of CANCEL_APPOINTMENT_SUCCESS
 */
export function updateAppointmentSuccess(updatedAppointment) {
  return {
    type: UPDATE_APPOINTMENT_SUCCESS,
    updatedAppointment,
  }
};

/**
 * Cancel an appointment encountered an error
 *
 * @param error - error that ocurred during cancelation
 *
 * @return {object}    An action object with a type of CANCEL_APPOINTMENT_ERROR
 */
export function updateAppointmentError(error) {
  return {
    type: UPDATE_APPOINTMENT_ERROR,
    error,
  }
};

/**
 * Decline an appointment
 *
 * @param id - appointment id
 * @param reason - reason for declining the appointment
 *
 * @return {object}    An action object with a type of DECLINE_APPOINTMENT
 */
export function declineAppointment(id, reason) {
  return {
    type: DECLINE_APPOINTMENT,
    id,
    reason,
  }
};

/**
 * Decline an appointment successfully
 *
 * @param updatedAppointment - newly updated appointment
 *
 * @return {object}    An action object with a type of DECLINE_APPOINTMENT_SUCCESS
 */
export function declineAppointmentSuccess(updatedAppointment) {
  return {
    type: DECLINE_APPOINTMENT_SUCCESS,
    id,
    reason,
  }
};

/**
 * Decline an appointment
 *
 * @param error - error that occurred when attempting to decline appointment
 *
 * @return {object}    An action object with a type of DECLINE_APPOINTMENT_ERROR
 */
export function declineAppointmentError(error) {
  return {
    type: DECLINE_APPOINTMENT_ERROR,
    error,
  }
};
