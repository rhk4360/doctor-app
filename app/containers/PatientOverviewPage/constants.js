export const GET_PATIENT_OVERVIEW = 'boilerplate/PatientOverview/GET_PATIENT_OVERVIEW';
export const GET_PATIENT_OVERVIEW_SUCCESS = 'boilerplate/PatientOverview/GET_PATIENT_OVERVIEW_SUCCESS';
export const GET_PATIENT_OVERVIEW_ERROR = 'boilerplate/PatientOverview/GET_PATIENT_OVERVIEW_ERROR';

export const CANCEL_APPOINTMENT = 'boilerplate/PatientOverview/CANCEL_APPOINTMENT';
export const UPDATE_APPOINTMENT_SUCCESS = 'boilerplate/PatientOverview/UPDATE_APPOINTMENT_SUCCESS';
export const UPDATE_APPOINTMENT_ERROR = 'boilerplate/PatientOverview/UPDATE_APPOINTMENT_ERROR';
export const DECLINE_APPOINTMENT = 'boilerplate/PatientOverview/DECLINE_APPOINTMENT';

