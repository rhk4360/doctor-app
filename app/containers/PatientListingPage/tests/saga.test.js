/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { GET_PATIENT_LIST } from '../constants';
import { getPatientsSuccess, getPatientsError } from '../actions';

import getPatientsData, { getPatientsList } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('getPatientsData Saga', () => {
  let getPatientsGenerator;
  const username = 'admin';
  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getPatientsGenerator = getPatientsList();

    const selectDescriptor = getPatientsGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getPatientsGenerator.next(username).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the getPatientsSuccess action if it requests the data successfully', () => {
    const response = [{
      name: 'Hello',
    }, {
      name: 'Good Bye',
    }];
    const putDescriptor = getPatientsGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(getPatientsSuccess(response, username)));
  });

  it('should call the getPatientsError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getPatientsGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(getPatientsError(response)));
  });
});

describe('getPatientsDataSaga Saga', () => {
  it('should start task to watch for GET_PATIENT_LIST action', () => {
    const takeLatestDescriptor = getPatientsData.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(GET_PATIENT_LIST, getPatientsData));
  });
});
