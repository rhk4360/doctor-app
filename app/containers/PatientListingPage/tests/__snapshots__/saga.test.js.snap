// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPatientsData Saga should call the getPatientsError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getPatientsData Saga should call the getPatientsError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3000/getProvidersPatients/undefined",
      Object {
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "method": "GET",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getPatientsData Saga should dispatch the getPatientsSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getPatientsData Saga should dispatch the getPatientsSuccess action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3000/getProvidersPatients/undefined",
      Object {
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "method": "GET",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getRepos Saga should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getRepos Saga should call the repoLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.github.com/users/mxstbr/repos?type=all&sort=updated",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getRepos Saga should dispatch the reposLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getRepos Saga should dispatch the reposLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.github.com/users/mxstbr/repos?type=all&sort=updated",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
